launcher/ui/pages/instance/InstanceSettingsPage.cpp

```diff
@@ -437,6 +452,67 @@ void InstanceSettingsPage::on_javaTestBtn_clicked()
     checker->run();
 }
 
+void InstanceSettingsPage::updateAccountsMenu()
+{
+    accountMenu->clear();
+
+    auto accounts = APPLICATION->accounts();
+    int accountIndex = accounts->findAccountByProfileId(m_settings->get("InstanceAccountId").toString());
+    MinecraftAccountPtr defaultAccount = accounts->defaultAccount();
+
+    if (accountIndex != -1 && accounts->at(accountIndex)) {
+        defaultAccount = accounts->at(accountIndex);
+    }
+
+    if (defaultAccount) {
+        ui->instanceAccountSelector->setText(defaultAccount->profileName());
+        ui->instanceAccountSelector->setIcon(defaultAccount->getFace());
```

Comment:
there is a small cornercase here regarding Offline accounts: Since they don't have an Icon, the returned `QPixmap` for the `getFace()` call is null. So, this doesn't set the icon the first time the page is opened and one of those is set. You'd probably need to do something like

```cpp
if (auto face = defaultAccount->getFace(); !face.isNull())
    ui->instanceAccountSelector->setIcon(face);
else
    ui->instanceAccountSelector->setIcon(APPLICATION->getThemedIcon("noaccount"));
```

for it to work as intended.

Considering this very same piece of code is repeated in other places here to, it may make sense to put this logic in a separate function :>

Reply:
Good thinking. Do you think this logic should be put in `MinecraftAccount`, named something like `getFaceIcon`?

Reply:
It can just be a simple function in `InstanceSettingsPage.cpp` i think, no need to get fancy, we don't use this logic anywhere else (i think?)

Reply:
Fixed! I also extracted the logic into a method called `getFaceForAccount` and used it where applicable :)